### DISCUSSION TITLE
Supports remote controls (e.g. RB4) for SP530e

### DISCUSSION THREAD

**Discussion by mik-laj** [monty68/uniled#84] [Ideas]
Hello,

I bought the SP530E controller with the RB4 remote control. Next, I installed WLED to control my LEDs, but later I realized that WLED does not support RB4.

I wonder how I can use RB4 now. The documentation says it's something at 2.4 GHz, but doesn't go into details. Has anyone looked at what protocol is used there?

It would be ideal to add support for Home Assistant, but if that wasn't possible then at least writing a module for ESPHome that would allow the signal to be passed to HA would be equally useful.

I disassembled the device and took photos of the circuits inside, which may be useful to us.

![IMG_20241219_054946587.jpg](https://github.com/user-attachments/assets/31cf4c1b-bbb1-42b2-a885-7cc74c706771)

![IMG_20241219_062714368.jpg](https://github.com/user-attachments/assets/d9925289-aedb-4eff-89bd-60786fadd9e5)

![IMG_20241219_062703550.jpg](https://github.com/user-attachments/assets/399faf3d-89af-4c46-8805-aa024c7ada20)

```
ST17H66B  
D2PD2326  
SHUJL923
```
I found a datasheet on Google. Here is a copy: https://drive.google.com/file/d/19xdyp2wp6p-PIa4VBV2ZbAqzqLEUBxqd/view?usp=drivesdk
This module is described as: ST17H66B2
Bluetooth Low Energy System on Chip with SIG-Mesh

![IMG_20241219_062651534.jpg](https://github.com/user-attachments/assets/2bfb3817-6c09-4cc0-b7d9-137981791a93)

![IMG_20241219_062657792.jpg](https://github.com/user-attachments/assets/58d41aa2-d0ae-4cb9-9ff8-8348b84b33bc)

```
CACHIP
CA51F351P6
S64012150C3U
```
https://en.zhwssz.com/produst15/290.html

On AliExpress we can find other remote controls, but I don't have access to them.
![Sf1887ae0ae9642e5976b349e829dbb90S.jpg_960x960q75.jpg_.jpg](https://github.com/user-attachments/assets/41bb0946-472a-49d4-b334-faa8451ded8d)

Thanks in advance for tips on how we can integrate this pilot with HA. I think there are more people in this project who have these devices, so I wrote here. I hope that together we will be able to develop something.

**Comment by gregb79** [comment]
Hi I know this is old, are you still looking for a solution  i just completed an ESP32-C3 ESPIDF gateway to MQTT and home assistant blueprint  to map buttons.
